src/project/internal/notationproject.cpp

```diff
@@ -168,7 +166,7 @@ mu::Ret NotationProject::doLoad(engraving::MscReader& reader, const io::path& st
 
     // Create new engraving project
     EngravingProjectPtr project = EngravingProject::create();
-    project->setPath(reader.params().filePath);
+    project->setFileInfoProvider(shared_from_this());
```

Comment:
There is a problem that I have fixed in this PR #10414
Passing a pointer to a child blocks deletion of the parent, because there are owners(child) of the parent

You should replace shared_from_this with weak_from_this

Reply:
In some situations, the file info provider should be "owned" by the engraving project, and in some situations not. 

For example, when creating a standalone QFileInfoProvider, it should be deleted when the EngravingProject is destroyed. 
But when we set the NotationProject as the info provider for the engraving project, the engraving project should obviously not delete it. 

I think a solution would be to introduce a new `class ProjectFileInfoProvider : public IFileInfoProvider` that will be passed to the engraving project, instead of passing the NotationProject itself. This way, the ProjectFileInfoProvider could be owned by the EngravingProject without problems; and the ProjectFileInfoProvider can hold a weak reference to the NotationProject. Does that sound like a good idea?